addons/jhub_cognito_auth.tf

```diff
@@ -0,0 +1,66 @@
+# ======================================
+# COGNITO
+
+variable "jhub_cognito_auth_enabled" {
+  default = false
+}
+
+variable "jhub_callback_url" {
+  default     = "https:///jhub.example.com/oauth_callback"
+  description = "the callback url for your jhub application"
+}
+
+resource "aws_cognito_user_pool" "pool" {
+  count = var.jhub_cognito_auth_enabled ? 1 : 0
+  name = "jhubuserpool"
+  alias_attributes           = ["email"]
+  auto_verified_attributes   = ["email"]
+
+  schema {
+    name                = "email"
+    attribute_data_type = "String"
+    mutable             = false
+    required            = true
+    string_attribute_constraints {
+      min_length = 1
+      max_length = 256
+    }
+  }
+
+  schema {
+    name                = "name"
+    attribute_data_type = "String"
+    mutable             = false
+    required            = true
+    string_attribute_constraints {
+      min_length = 1
+      max_length = 256
+    }
+  }
+  password_policy {
+    minimum_length    = 8
+    require_uppercase = true
+    require_lowercase = true
+    require_numbers   = true
+    require_symbols   = false
+  }
+}
+
+resource "aws_cognito_user_pool_client" "client" {
```

Comment:
Can we add https://www.terraform.io/docs/configuration/resources.html#prevent_destroy as an optional variable?

Reply:
I don't seem to be able to use variables in the lifecycle block:
https://github.com/hashicorp/terraform/issues/10730

Reply:
Fair enough, let's just leave it out for the moment then.